bound/kt/src/main/kotlin/web5/sdk/dids/DidResolutionResult.kt

```diff
@@ -0,0 +1,144 @@
+package web5.sdk.dids
+
+import web5.sdk.rust.DocumentMetadataData as RustCoreDocumentMetadataData
+import web5.sdk.rust.ResolutionMetadataData as RustCoreResolutionMetadataData
+import web5.sdk.rust.ResolutionMetadataError as RustCoreResolutionMetadataError
+import web5.sdk.rust.ResolutionResultData as RustCoreResolutionResultData
+
+/**
+ * Representation of the result of a DID (Decentralized Identifier) resolution.
+ *
+ * @property didDocument The resolved DID document, if available.
+ * @property didDocumentMetadata The metadata associated with the DID document.
+ * @property didResolutionMetadata The metadata associated with the DID resolution process.
+ */
+data class DidResolutionResult(
+    val didDocument: DidDocument? = null,
+    val didDocumentMetadata: DidDocumentMetadata? = null,
+    val didResolutionMetadata: DidResolutionMetadata = DidResolutionMetadata()
```

Comment:
technically in the APID the `did*` prefix on these data member names is not included, so just `document` or `documentMetadata` https://github.com/TBD54566975/web5-rs/blob/kendall/rfc-0001/docs/API_DESIGN.md#resolutionresult

I do think we'll need consistency in the name, albeit different casing, in order to be considered conformant to the APID. we could change the APID tho, I'm not opposed

Reply:
Yea I could go either way, I noticed it was just "Document" Seemed really generic, but I'll stick with it! 

I'll drop the Did on these to go towards the APID for now, and we can chnage later if we change APID